docs/messaging.md

```diff
@@ -0,0 +1,58 @@
+# Messaging API
+
+## Messaging Request
+
+Services (such as A3) can use Thentos to send messages to users or arbitrary
+email addresses. To send a message, the service sends a POST request with
+JSON payload to the `THENTOS_URL/message` endpoint. The payload must be a
+JSON object with the following required fields:
+
+* subject: the subject of the message
+* body: the plain-text body of the message
+
+The following field is optional:
+
+* html: the HTML-formatted body of the message
+
+The following fields are alternative--exactly one of them must be present:
+
+* target_users: a list of one or more strings where each string is the user
+  path of a user. The message will be send to each of these users.
+  Implementation note: *users* in a service (such as A3) are *personas* in
+  Thentos. In Thentos, a service's notation of a user path is stored as
+  *ExternalUrl* of the persona.
+* target_email: a list of one or more strings where each string is an email
```

Comment:
`target_emails`?  or singular in both cases and accept both a single value and a list as argument?  i prefer the latter.